SDCTA/src/components/auth/SignUpPage.tsx

```diff
@@ -24,149 +14,15 @@ export const SignUpPage: React.FC = () => {
-        // if (error.code === 'auth/account-exists-with-different-credential') {
-        //   // The pending Google credential.
-        //   var pendingCred = error.credential;
-        //   // The provider account's email address.
-        //   var email = error.email;
-        //   // Get sign-in methods for this email.
-        //   auth_.fetchSignInMethodsForEmail(email).then(function(methods) {
-        //     // If the user has several sign-in methods,
-        //     // the first method in the list will be the "recommended" method to use.
-        //     if (methods[0] === 'password') {
-        //       // Asks the user their password.
-        //       // TODO: handle this asynchronously.
-        //       var password = promptUserForPassword(); // TODO: implement promptUserForPassword.
-        //       auth_.signInWithEmailAndPassword(email, password).then(function(result) {
-        //         return result.user.linkWithCredential(pendingCred);
-        //       }).then(function() {
-        //         // Google account successfully linked to the existing Firebase user.
-        //         // goToApp();
-        //         navigate("/");
-        //       });
-        //       return;
-        //     }
-        //     // All the other cases are external providers.
-        //     // Construct provider object for that provider.
-        //     // TODO: implement getProviderForProviderId.
-        //     var provider = getProviderForProviderId(methods[0]);
-        //     // At this point, you should let the user know that they already have an account
-        //     // but with a different provider, and let them validate the fact they want to
-        //     // sign in with this provider.
-        //     // Sign in to provider. Note: browsers usually block popup triggered asynchronously,
-        //     // so in real scenario you should ask the user to click on a "continue" button
-        //     // that will trigger the signInWithPopup.
-        //     auth_.signInWithPopup(provider).then(function(result) {
-        //       // Remember that the user may have signed in with an account that has a different email
-        //       // address than the first one. This can happen as Firebase doesn't control the provider's
-        //       // sign in flow and the user is free to login using whichever account they own.
-        //       // Link to Google credential.
-        //       // As we have access to the pending credential, we can directly call the link method.
-        //       result.user.linkAndRetrieveDataWithCredential(pendingCred).then(function(usercred) {
-        //         // Google account successfully linked to the existing Firebase user.
-        //         // goToApp();
-        //         navigate("/");
-        //       });
-        //     });
-        //   });
-        // }
-      }); // end of catch
+  const callSignupWithGoogle = (): void => {
+    dispatch({
+      type: "SIGNUP_GOOGLE_USER",
```

Comment:
We probably want action strings kept in a constants file or more likely just the file where the slice is defined